spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkCopyOnWriteOperation.java

```diff
@@ -64,7 +71,7 @@ public Command command() {
   public ScanBuilder newScanBuilder(CaseInsensitiveStringMap options) {
     if (lazyScanBuilder == null) {
       lazyScanBuilder =
-          new SparkScanBuilder(spark, table, options) {
+          new SparkScanBuilder(spark, table, Spark3Util.buildScanOptions(branch, options)) {
```

Comment:
Looking at this class, passing `branch` to the `SparkScanBuilder` would also make the read and write paths more similar.